File: python/mathAlgos.py
def fib(n):
    if n < 1:
        return 'Input must be positive'

    nMinusOne = 1
    nMinusTwo = 0

    for i in range(2, n):
        nMinusTwo, nMinusOne = nMinusOne, nMinusOne + nMinusTwo

    return nMinusOne + nMinusTwo

def fac(n):
    if n < 0:
        return 'invalid input'

    product = 1

    for i in reversed(range(2, n + 1)):
        product *= i

    return product

File: javascript/mathAlgos.js
function fib(n) {
    if (n < 1) return 'invalid input'

    let nMinusOne = 1;
    let nMinusTwo = 0;

    for (let i = 2; i < n; i++) {
        const temp = nMinusTwo;
        nMinusTwo = nMinusOne;
        nMinusOne += temp;
    }

    return nMinusOne + nMinusTwo;
}

for (let i = -1; i < 10; i++) {
    console.log(fib(i));
}
